src/mac-now-playing/MacNowPlaying.mm

```diff
@@ -0,0 +1,267 @@
+    commands.disableLanguageOptionCommand.enabled = NO;
+    commands.ratingCommand.enabled = NO;
+    commands.likeCommand.enabled = NO;
+    commands.dislikeCommand.enabled = NO;
+    commands.bookmarkCommand.enabled = NO;
+}
+
+static void remote_command_set_playback_state()
+{
+    MPNowPlayingInfoCenter * center = [MPNowPlayingInfoCenter defaultCenter];
+    MPNowPlayingPlaybackState state = MPNowPlayingPlaybackStateStopped;
+    if (aud_drct_get_playing())
+    {
+        bool paused = aud_drct_get_paused();
+        state = paused ? MPNowPlayingPlaybackStatePaused
+                       : MPNowPlayingPlaybackStatePlaying;
+    }
+    [center setPlaybackState:state];
+}
+
+static void remote_command_start()
+{
+    MPNowPlayingInfoCenter * center = [MPNowPlayingInfoCenter defaultCenter];
+    // Little hack to get the now playing starting otherwise audacious
+    // don't react to media keys
+    [center setPlaybackState:MPNowPlayingPlaybackStatePaused];
+    [center setPlaybackState:MPNowPlayingPlaybackStatePlaying];
+    remote_command_set_playback_state();
+    update_metadata(nullptr, nullptr);
+}
+
+static void update_position(void * data, void * user) { update_player_time(); }
+
+static void update_metadata(void * data, void * user)
+{
+    MPNowPlayingInfoCenter * center = [MPNowPlayingInfoCenter defaultCenter];
+    if (aud_drct_get_ready())
+    {
+        Tuple tuple = aud_drct_get_tuple();
+        ::String title = tuple.get_str(Tuple::Title);
+        String artist = tuple.get_str(Tuple::Artist);
+        String album = tuple.get_str(Tuple::Album);
+        String file = aud_drct_get_filename();
+        int current_time = aud_drct_get_time();
+        int length = tuple.get_int(Tuple::Length);
+
+        NSMutableDictionary<NSString *, id> * nowPlayingInfo =
+            [[NSMutableDictionary alloc] init];
+
+        nowPlayingInfo[MPNowPlayingInfoPropertyMediaType] =
+            [[NSNumber alloc] initWithInt:MPNowPlayingInfoMediaTypeAudio];
+        nowPlayingInfo[MPNowPlayingInfoPropertyPlaybackRate] =
+            [[NSNumber alloc] initWithBool:aud_drct_get_playing()];
+        nowPlayingInfo[MPMediaItemPropertyTitle] =
+            [[NSString alloc] initWithUTF8String:title ? title : ""];
+        nowPlayingInfo[MPMediaItemPropertyArtist] =
+            [[NSString alloc] initWithUTF8String:artist ? artist : ""];
+        nowPlayingInfo[MPMediaItemPropertyAlbumTitle] =
+            [[NSString alloc] initWithUTF8String:album ? album : ""];
+        nowPlayingInfo[MPMediaItemPropertyMediaType] =
+            [[NSNumber alloc] initWithInt:MPMediaTypeMusic];
+        nowPlayingInfo[MPMediaItemPropertyPlaybackDuration] =
+            [[NSNumber alloc] initWithInt:length / 1000];
+        nowPlayingInfo[MPNowPlayingInfoPropertyElapsedPlaybackTime] =
+            [[NSNumber alloc] initWithInt:current_time];
+
+        if (file)
+        {
+            AudArtPtr artwork = aud_art_request(file, AUD_ART_FILE);
```

Comment:
Use `AUD_ART_DATA` instead. Then also local cover files (e.g. Cover.jpg stored in the same dir as the audio file) are shown.